dom.go

```diff
@@ -991,19 +994,40 @@ func renderComponent(next Component, prev ComponentOrHTML) (h *HTML, skip bool,
 
 	// Render the component into HTML, handling nil renders.
 	nextRender := next.Render()
+	prevRender := next.Context().prevRender
 	if nextRender == nil {
 		// nil renders are translated into noscript tags.
 		nextRender = Tag("noscript")
 	}
 
-	// Reconcile the actual rendered HTML.
-	pendingMounts = nextRender.reconcile(extractHTML(prev))
+	switch v := nextRender.(type) {
+	case Component:
+		nextHTML, skip, pendingMounts = renderComponent(v, prevRender)
```

Comment:
We recursively render components until we hit HTML, then bubble that up to the originating component.